pkg/addons/adminconsole/adminconsole.go

```diff
@@ -330,10 +323,47 @@ func createRegistrySecret(ctx context.Context, cli client.Client, namespace stri
 		},
 		Type: "kubernetes.io/dockerconfigjson",
 	}
+
 	err := cli.Create(ctx, &registryCreds)
 	if err != nil {
 		return fmt.Errorf("unable to create registry-auth secret: %w", err)
 	}
 
 	return nil
 }
+
+func createKotsPasswordSecret(ctx context.Context, cli client.Client, namespace string, password string) error {
+	if err := kubeutils.WaitForNamespace(ctx, cli, namespace); err != nil {
+		return err
+	}
+
+	passwordBcrypt, err := bcrypt.GenerateFromPassword([]byte(password), 10)
+	if err != nil {
+		return fmt.Errorf("unable to create kotsadm-password secret: %w", err)
```

Comment:
I think this error message could be more specific. The "unable to create secret" would wrap this.